File: pset6/mario/mario.py

from cs50 import get_int

while True:
    n = get_int("Height:")
    if n >= 0 and n <= 23:
        break

for i in range(n):
    for j in range(n-1-i):
        print(" ",end="")
    for k in range(i+1):
        print("#",end="")
    print("  ",end="")
    for h in range(i+1):
        print("#",end="")
    print("")

File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 0 || n > 23);

    //draw the half-pyramid
    for (int i = 0; i < n ; i++)
    {
        //print spaces
        for (int j = 0; j < n - 1 - i ; j++)
        {
            printf(" ");
        }
        //print hashes
        for (int k = 0; k < i + 2; k++)
        {
            printf("#");
        }
        //print new line
        printf("\n");
    }
}